### ISSUE TITLE
Failed to upload the chunk ... failed to send packet header: EOF

### ISSUE THREAD

**Issue by vibl** [gilbertchen/duplicacy#406]
There's an error message when I run `duplicacy backup -stats`:
```
Failed to upload the chunk 2e9ce7aeaa0cc4e077d38d527999ac33cf90786cb9a753d69637ad8d35a14e9a: 
failed to send packet header: EOF
```
This error message appears after a few minutes and after the messages:
```
Storage set to sftp://abc.def...
Last backup at revision 3 found
Indexing /ghi/jkl
Discarding file attributes
```

Some information that might be useful:
- The command was running fine until now.
- I can access the SFTP server from the backed up machine.
- I interrupted a backup a few hours earlier.
- There are about 9 millions files to backup, and this number have increased by 1 or 2 millions since the last backup.
- I already tried setting "DUPLICACY_ATTRIBUTE_THRESHOLD=10000000" in case it would solve a possible timeout problem (it didn't).

**Comment by gilbertchen** [comment]
`Discarding file attributes` isn't an error.  It just means that file attributes are removed from the file list to save memory and they will be read again when uploading the file list as a part of the final snapshot file.

The EOF error was caused by a broken network connection.  There isn't a retry mechanism in the sftp backend (we should probably have).

**Comment by jwatt** [comment]
Are there any plans to implement this in the near-term?

**Comment by archon810** [comment]
Found this thread because I am starting a large backup for which permissions/ownership info are important and saw a disturbing `Discarding file attributes` message.

Looks like the data will be preserved after all, but may I suggest tweaking this message to make it less ominous and a bit more informative (don't worry, the file attributes will be recorded at a later stage), or removing it entirely?

Thanks.

**Comment by sagar-srivastava** [comment]
I agree. sftp backend is pretty much useless without a retry/timeout mechanism. backup stops in the middle and never starts again.
